cmd/snap/cmd_userd.go

```diff
@@ -74,3 +79,11 @@ func (x *cmdUserd) Execute(args []string) error {
 
 	return x.userd.Stop()
 }
+
+func (x *cmdUserd) runAutostart() error {
+	err := userd.AutostartSessionApps()
+	if err != nil {
+		return fmt.Errorf("autostart failed for the following apps:\n%v", err)
```

Comment:
How about `cannot auto-start some applications: %s` (note the lack of `\n`)

Reply:
That's a case when the error can take multiple lines, because a number of desktop files may fail to be autostarted each with potentially different error, eg. (as seen https://github.com/snapcore/snapd/pull/4942/files#diff-be4d73e8c4c228314eda9696a8205ebfR247):
```
- "foo-stable.desktop": cannot determine startup command: Exec not found or invalid
- "no-match.desktop": cannot match desktop file with snap "b-foo" applications
- "no-snap.desktop": cannot obtain snap information for snap "c-foo": cannot find current revision.*no such file or directory
```

Current format will correctly print:
```
autostart failed for the following apps:
- "foo-stable.desktop": cannot determine startup command: Exec not found or invalid
...
```
IIRC I saw a similar pattern in the code elsewhere, hence the formatting.